File: others/프로그래머스_완주하지못한선수.py
def solution(participant, completion):

    completed = {}

    # 참가자 인원 세기
    for each in participant:
        completed[each] = completed.get(each, 0) + 1

    # 완주자 인원 빼기
    for each in completion:
        completed[each] -= 1

        # 모두 다 카운트 끝났으면 리스트에서 없애버린다.
        if completed[each] == 0:
            del completed[each]

    return list(completed.keys())[0]

File: others/프로그래머스_완주하지못한선수.js
function solution(participant, completion) {

  const participated = Object

  participant.forEach(each => {
    if (each in participated)
      participated[each] += 1
    else participated[each] = 1
  })

  completion.forEach(each => {
    participated[each] -= 1
    if (participated[each] === 0) {
      delete participated[each]
    }
  })

  return Object.keys(participated)[0]
}

solution(["leo", "kiki", "eden"], ["eden", "kiki"])
